frontend/web/components/CollapsibleNestedList.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useState, forwardRef, useImperativeHandle } from 'react'
+import Icon from 'components/Icon'
+import { EditPermissionsModal } from 'components/EditPermissions'
+import {
+  useGetRoleProjectPermissionsQuery,
+  useGetRoleEnvironmentPermissionsQuery,
+} from 'common/services/useRolePermission'
+import Format from 'common/utils/format'
+
+type MainItem = {
+  name: string
+  id: string
+}
+
+type CollapsibleNestedListProps = {
+  mainItems: MainItem[]
+  role: Role
+  level: string
+}
+
+const CollapsibleNestedList: React.FC<CollapsibleNestedListProps> = forwardRef(
```

Comment:
Not really sure if this is the right name for this, if it's coupled with projects and permissions

Reply:
Renamed.